forte/data/multi_pack.py

```diff
@@ -176,6 +176,172 @@ def get_span_text(self, begin: int, end: int):
             "specific data pack to get text."
         )
 
+    def remove_pack(
```

Comment:
There are a few iteration functions in the multi pack, which can be updated so that even if we don't do `pruning`, there will still be no `None` packs coming out from them 

https://github.com/asyml/forte/blob/9f7bbdc2188c8659eea44872446111dc58499fb6/forte/data/multi_pack.py#L467
https://github.com/asyml/forte/blob/9f7bbdc2188c8659eea44872446111dc58499fb6/forte/data/multi_pack.py#L479
https://github.com/asyml/forte/blob/9f7bbdc2188c8659eea44872446111dc58499fb6/forte/data/multi_pack.py#L486